### PR TITLE
setup.sh - Automating our environment

### PR DESCRIPTION
**Pull request by MeitarRizner**
Adds commands that are executed automatically when doing 'vagrant up'.
These commands doing:
● Installing the Pipenv tool.
● Telling Pipenv to install our software dependencies.
● Starting our application.

### CODE REVIEW

File: BuyTogether/settings.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+# Application definition
+
+INSTALLED_APPS = [
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
```
Comment (pazMenachem) [inline BuyTogether/settings.py:39]: missing our app installation here or is it in some one else task?

Comment (thetaxas2468) [reply BuyTogether/settings.py:39]: I will update this on my PR dw.

**Review by LiorSilberman — COMMENTED** [review]
looks good for me!

**Comment by thetaxas2468** [comment]
> looks good for me!

there is something missing ill be adding it in my PR, buyTogether.apps.buyTogetherConfig

**Comment by brochwerger** [comment]
Is this PR about changes to bootstrap.sh ? If yes, then the commits with changes to other files don't belong. If not, it is badly named. Please review and either break it into different PRs (my preference) of be more specific explaining what changes this PR includes

File: setup.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# password if we don't have sufficient privileges to run it
+sudo -n dnf install -y pipenv
+
+cd /vagrant
+# Install dependencies with Pipenv
+pipenv sync --dev
+
+# run our app. Nohup and "&" are used to let the setup script finish
+# while our app stays up. The app logs will be collected in nohup.out
+nohup pipenv run python manage.py runserver 0.0.0.0:8000 &
```
Comment (yftacherzog) [inline setup.sh:15]: See that red symbol below the last line. It tells you, there's no newline at the end of the file. You should generally have a single newline at the end of the file (no more, no less). Most editors also have configurations for adding that when  files are saved, so you don't have to worry about it.

image

File: Vagrantfile [hunk 3]
```diff
@@ -16,6 +16,8 @@ Vagrant.configure("2") do |config|
 
   config.vm.synced_folder ".", "/vagrant", type: "virtualbox"
   
+  config.vm.provision "shell", path: "setup.sh", privileged: false
```
Comment (yftacherzog) [inline Vagrantfile:19]: The PR description and commit message mention `bootstrap.sh` while you actually use `setup.sh`. I don't think it matters which one you use, but make sure the commit message, PR description and actual changes match.

**Comment by yftacherzog** [comment]
@MeitarRizner Note that you have 2 commits now. They need to be squashed.

File: Vagrantfile (hunk 3 above)
Comment (yftacherzog) [reply Vagrantfile:19]: please rename the PR accordingly as well

Comment (yftacherzog) [reply Vagrantfile:19]: @MeitarRizner please address this comment.

**Pull request merged**